src/models/local/LocalBaseController.js

```diff
@@ -92,6 +97,26 @@ export default class LocalBaseController {
     return !!abi.find(fn => fn.type === "constructor");
   }
 
+  hasSelfDestruct(contractDataPath) {
+    if (!fs.exists(contractDataPath)) return false
+    const ast = fs.parseJson(contractDataPath).ast
+    return this.hasKeyValue(ast, "typeIdentifier", "t_function_selfdestruct_nonpayable$_t_address_$returns$__$");
+  }
+
+  hasKeyValue(data, key, value) {
+    if (!data) return false
+    if (data[key] === value) return true
+    if (typeof(data) === 'object') {
+      for (const child_key in data) {
```

Comment:
Please make sure to use `camelCase` for all variable names, and not `snake_case`, for consistency